sklearn/semi_supervised/label_propagation.py

```diff
@@ -264,13 +280,20 @@ def fit(self, X, y):
             l_previous = self.label_distributions_
             self.label_distributions_ = safe_sparse_dot(
                 graph_matrix, self.label_distributions_)
+
+            if self.variant == 'propagation':
+                normalizer = np.sum(
+                    self.label_distributions_, axis=1)[:, np.newaxis]
+                self.label_distributions_ /= normalizer
+
             # clamp
             self.label_distributions_ = np.multiply(
                 clamp_weights, self.label_distributions_) + y_static
```

Comment:
Yes, `np.where(unlabeled[:, np.newaxis], self.label_distributions_, y_static)` to be more precise. Have factored out `unlabeled[:, np.newaxis]` in the actual implementation.